jwcrypto/jwe.py

```diff
@@ -289,6 +290,56 @@ def unwrap(self, key, keylen, ek, headers):
         return cek
 
 
+class _Pbes2HsAesKw(_RawKeyMgmt):
+
+    def __init__(self, hashsize, keysize):
+        self.backend = default_backend()
+        self.hashsize = hashsize
+        self.keysize = keysize // 8
+
+    def _get_key(self, alg, key, p2s, p2c):
```

Comment:
This is your library and ultimately you need to choose. But the above is my preference.